### PR TITLE
Add welford operation to python-frontend

### PR DESCRIPTION
**Pull request by rdspring1**
This PR adds `welford` operation to python frontend and support for it in `FusionTranslation`.

For example, the `variance_mean` composite operation creates `WelfordOp` expression. For cpp to python translation, we need to map the `WelfordOp` expression back to its corresponding python operation.

### CODE REVIEW

**Comment by rdspring1** [comment]
!build

File: tests/python/test_python_frontend.py [hunk 1]
```diff
@@ -448,7 +448,7 @@ def nvfuser_fusion(
             eps=1e-12,
             keepDim=True,
         )
-        nvf_out, _ = self.exec_nvfuser(fusion_func, inputs)
+        nvf_out, _ = self.exec_nvfuser(fusion_func, inputs, is_clonable=True)
```
Comment (jjsjann123) [inline tests/python/test_python_frontend.py:451]: qq: this function doesn't use welford, is this just catching something missed out earlier?

**Review by jjsjann123 — APPROVED** [review]
LGTM.

File: csrc/python_frontend/python_bindings.cpp [hunk 2]
```diff
@@ -3130,7 +3130,30 @@ void initNvFuserPythonBindings(PyObject* module) {
...
+        size_t ndims = (arg.dims - axes.size());
+        Tensor avg = fd->defineTensor(ndims);
+        Tensor var_sum = fd->defineTensor(ndims);
+        Tensor n = fd->defineTensor(ndims);
+        fd->defineRecord(new WelfordOpRecord(
+            {fd->recordingState(arg())},
+            {fd->recordingState(avg()),
+             fd->recordingState(var_sum()),
+             fd->recordingState(n())},
+            std::move(axes)));
```
[the remark this answers is not in the document]
Comment (rdspring1) [reply csrc/python_frontend/python_bindings.cpp]: Removed

**Comment by rdspring1** [comment]
!build

**Comment by rdspring1** [comment]
!build

File: csrc/python_frontend/python_bindings.cpp [hunk 3]
```diff
@@ -3130,7 +3130,30 @@ void initNvFuserPythonBindings(PyObject* module) {
       py::arg("correction") = 1,
       py::arg("keepdim") = false,
       py::return_value_policy::reference);
-
+  nvf_ops.def(
+      "welford",
+      [](FusionDefinition::Operators& self,
+         Tensor arg,
+         const std::vector<int64_t>& dims) -> decltype(auto) {
```
Comment (jjsjann123) [inline csrc/python_frontend/python_bindings.cpp:3137]: :clap:

**Pull request merged**